web/src/hashchange.js

```diff
@@ -345,7 +352,17 @@ function do_hashchange_overlay(old_hash) {
 
     if (base === "groups") {
         const right_side_tab = hash_parser.get_current_nth_hash_section(3);
-        user_group_edit.launch(section, right_side_tab);
+
+        if (is_somebody_else_profile_open()) {
+            user_group_edit.launch(section, "all-groups", right_side_tab);
+            return;
+        }
```

Comment:
It feels somewhat unclean to me to be doing this profile check here; I think it would be more cleanly done by registering a click handler for these group elements rather than relying solely on the URL, and using that to pass through a parameter. But doing that might be a pain, so I'm not sure it's worth blocking this PR on that.